### DISCUSSION TITLE
Error downloading ERA5 models (website update)

### DISCUSSION THREAD

**Discussion by fgdiazm** [insarlab/MintPy#1294] [Q&A]
Hi everyone

I processed some Sentinel-1 images with ISCE2, and when processing them in MINTPY, with the pyaps option, I can't download the ERA5 models, because there is an update to the website and therefore the link is outdated.

------------------------------------------------------------------------------
downloading weather model data using PyAPS ...
number of grib files to download: 94
------------------------------------------------------------------------------

INFO: You are using the latest ECMWF platform for downloading datasets: https://cds.climate.copernicus.eu/api/v2
Downloading 1 of 94: /home/work/geored-insar/Isce2/Sabana_Bogota/DSC/ZT_InSAR_561-2024_villao_oct2014_marz2024_NE/MInt                                                                     py_try1/ERA5/ERA5_N0_N10_W80_W60_20141029_11.grb
{'product_type': 'reanalysis', 'format': 'grib', 'variable': ['geopotential', 'temperature', 'specific_humidity'], 'pr                                                                     essure_level': ['1', '2', '3', '5', '7', '10', '20', '30', '50', '70', '100', '125', '150', '175', '200', '225', '250'                                                                     , '300', '350', '400', '450', '500', '550', '600', '650', '700', '750', '775', '800', '825', '850', '875', '900', '925                                                                     ', '950', '975', '1000'], 'year': '2014', 'month': '10', 'day': '29', 'time': '11:00', 'area': '10.00/-80.00/0.00/-60.                                                                     00'}
2024-11-12 20:06:45,702 INFO Sending request to https://cds.climate.copernicus.eu/api/v2/resources/reanalysis-era5-pre                                                                     ssure-levels
**WARNING: the 1 attampt to download failed, retry it.**

INFO: You are using the latest ECMWF platform for downloading datasets: https://cds.climate.copernicus.eu/api/v2
Downloading 1 of 94: /home/work/geored-insar/Isce2/Sabana_Bogota/DSC/ZT_InSAR_561-2024_villao_oct2014_marz2024_NE/MInt                                                                     py_try1/ERA5/ERA5_N0_N10_W80_W60_20141029_11.grb
{'product_type': 'reanalysis', 'format': 'grib', 'variable': ['geopotential', 'temperature', 'specific_humidity'], 'pr                                                                     essure_level': ['1', '2', '3', '5', '7', '10', '20', '30', '50', '70', '100', '125', '150', '175', '200', '225', '250'                                                                     , '300', '350', '400', '450', '500', '550', '600', '650', '700', '750', '775', '800', '825', '850', '875', '900', '925                                                                     ', '950', '975', '1000'], 'year': '2014', 'month': '10', 'day': '29', 'time': '11:00', 'area': '10.00/-80.00/0.00/-60.                                                                     00'}
2024-11-12 20:06:46,141 INFO Sending request to https://cds.climate.copernicus.eu/api/v2/resources/reanalysis-era5-pre                                                                     ssure-levels
**WARNING: the 2 attampt to download failed, retry it.**

INFO: You are using the latest ECMWF platform for downloading datasets: https://cds.climate.copernicus.eu/api/v2
Downloading 1 of 94: /home/work/geored-insar/Isce2/Sabana_Bogota/DSC/ZT_InSAR_561-2024_villao_oct2014_marz2024_NE/MInt                                                                     py_try1/ERA5/ERA5_N0_N10_W80_W60_20141029_11.grb
{'product_type': 'reanalysis', 'format': 'grib', 'variable': ['geopotential', 'temperature', 'specific_humidity'], 'pr                                                                     essure_level': ['1', '2', '3', '5', '7', '10', '20', '30', '50', '70', '100', '125', '150', '175', '200', '225', '250'                                                                     , '300', '350', '400', '450', '500', '550', '600', '650', '700', '750', '775', '800', '825', '850', '875', '900', '925                                                                     ', '950', '975', '1000'], 'year': '2014', 'month': '10', 'day': '29', 'time': '11:00', 'area': '10.00/-80.00/0.00/-60.                                                                     00'}
2024-11-12 20:06:46,548 INFO Sending request to https://cds.climate.copernicus.eu/api/v2/resources/reanalysis-era5-pre                                                                     ssure-levels

**************************************************
**WARNING: downloading failed for 3 times, stop trying and continue.**
**************************************************

------------------------------------------------------------------------------
calculating absolute delay for each date using PyAPS (Jolivet et al., 2011; 2014) ...
number of grib files used: 0
Traceback (most recent call last):
  File "/home/ops/tools/MintPy/mintpy/smallbaselineApp.py", line 1288, in <module>
    main(sys.argv[1:])
  File "/home/ops/tools/MintPy/mintpy/smallbaselineApp.py", line 1270, in main
    app.run(steps=inps.runSteps)
  File "/home/ops/tools/MintPy/mintpy/smallbaselineApp.py", line 1059, in run
    self.run_tropospheric_delay_correction(sname)
  File "/home/ops/tools/MintPy/mintpy/smallbaselineApp.py", line 811, in run_tropospheric_delay_correction
    tropo_pyaps3.main(iargs)
  File "/home/ops/tools/MintPy/mintpy/tropo_pyaps3.py", line 836, in main
    calc_delay_timeseries(inps)
  File "/home/ops/tools/MintPy/mintpy/tropo_pyaps3.py", line 759, in calc_delay_timeseries
    prog_bar = ptime.progressBar(maxValue=num_date, print_msg=~inps.verbose)
  File "/home/ops/tools/MintPy/mintpy/objects/progress.py", line 52, in __init__
    self.reset()
  File "/home/ops/tools/MintPy/mintpy/objects/progress.py", line 57, in reset
    self.update_amount(0)  # Build progress bar string
  File "/home/ops/tools/MintPy/mintpy/objects/progress.py", line 73, in update_amount
    percentDone = (diffFromMin / np.float(self.span)) * 100.0
ZeroDivisionError: float division by zero
(mintpy) ops@49e3fec00469:/home/work/geored-insar/Isce2/Sabana_Bogota/DSC/ZT_InSAR_561-2024_villao_oct2014_marz2024_NE                                                                     /MIntpy_try1$ smallbaselineApp.py 20241104_smallbaseline_REVE_FREDY_VILLAO_DSC_try1_ERA5_oct2014-sep2024.cfg

___________________________________________________________

How can I fix this so I can continue with the processing?

**Comment by zhdqaq** [comment]
i met this error too

**Comment by barely-sad-one** [comment]
I have Same error.

**Comment by yunjunz** [comment]
You are not using the latest PyAPS. Please check its website for the latest code and account setup info.
